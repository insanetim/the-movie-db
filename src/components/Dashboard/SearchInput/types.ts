export interface SearchInputProps {
  query: string
}

export interface SearchInputHookProps extends SearchInputProps {}

export interface SearchInputHook {
  currentValue: string
  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  handleSearch: (value: string) => void
}
