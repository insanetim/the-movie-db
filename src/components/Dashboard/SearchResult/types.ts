import {
  ErrorMsg,
  HandlePagination,
  Query,
} from 'src/interfaces/global.interface'
import { IMoviesList } from 'src/interfaces/movie.interface'

export type SearchResultHookProps = SearchResultProps

export type SearchResultHookReturn = {
  error: ErrorMsg
  handlePagination: HandlePagination
  isLoading: boolean
  movies: IMoviesList | undefined
}

export type SearchResultProps = {
  query: Query
}
