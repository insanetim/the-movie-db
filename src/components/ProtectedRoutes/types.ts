import { Location } from 'react-router-dom'

export type ProtectedRoutesHook = {
  isAuthenticated: boolean
  location: Location
}
