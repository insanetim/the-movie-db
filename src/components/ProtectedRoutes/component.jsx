import { Outlet, Navigate } from 'react-router-dom'

import useContainer from './hook'

const ProtectedRoutes = () => {
  const { sessionId, location } = useContainer()

  return sessionId ? (
    <Outlet />
  ) : (
    <Navigate
      to='/login'
      replace
      state={{ from: location }}
    />
  )
}

export default ProtectedRoutes
