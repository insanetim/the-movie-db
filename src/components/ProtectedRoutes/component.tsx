import { Navigate, Outlet } from 'react-router-dom'

import useContainer from './hook'

const ProtectedRoutes: React.FC = () => {
  const { isAuthenticated, location } = useContainer()

  return isAuthenticated ? (
    <Outlet />
  ) : (
    <Navigate
      replace
      state={{ from: location }}
      to='/login'
    />
  )
}

export default ProtectedRoutes
