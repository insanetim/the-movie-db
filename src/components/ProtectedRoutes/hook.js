import { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'

import { sessionIdSelector } from 'src/state/session/selectors'
import { fetchAccount } from 'src/state/session/actions'

const useContainer = () => {
  const dispatch = useDispatch()
  const sessionId = useSelector(sessionIdSelector)

  useEffect(() => {
    if (sessionId) {
      dispatch(fetchAccount())
    }
  }, [sessionId])

  return { sessionId }
}

export default useContainer
