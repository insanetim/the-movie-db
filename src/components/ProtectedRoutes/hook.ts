import { useLocation } from 'react-router-dom'
import { useAppSelector } from 'src/hooks/useRedux'
import { sessionIdSelector } from 'src/store/auth/selectors'

import { ProtectedRoutesHook } from './types'

const useContainer = (): ProtectedRoutesHook => {
  const sessionId = useAppSelector(sessionIdSelector)
  const location = useLocation()

  return { location, sessionId }
}

export default useContainer
