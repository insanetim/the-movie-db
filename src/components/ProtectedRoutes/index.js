import ProtectedRoutes from './component'

export default ProtectedRoutes
