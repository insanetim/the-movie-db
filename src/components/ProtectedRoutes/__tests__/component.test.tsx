import { Location } from 'react-router-dom'
import renderWithWrapper from 'src/utils/testHelpers/renderWithWrapper'

import ProtectedRoutes from '../component'
import { ProtectedRoutesHook } from '../types'

const mockedHook: ProtectedRoutesHook = {
  isAuthenticated: true,
  location: {} as Location,
}
jest.mock('../hook', () => jest.fn(() => mockedHook))

describe('ProtectedRoutes component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<ProtectedRoutes />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot without isAuthenticated', () => {
    mockedHook.isAuthenticated = false
    const { asFragment } = renderWithWrapper(<ProtectedRoutes />)

    expect(asFragment()).toMatchSnapshot()
  })
})
