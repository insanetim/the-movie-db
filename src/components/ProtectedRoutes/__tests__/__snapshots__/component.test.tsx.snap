// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProtectedRoutes component should match snapshot 1`] = `<DocumentFragment />`;

exports[`ProtectedRoutes component should match snapshot without sessionId 1`] = `<DocumentFragment />`;
